### ISSUE TITLE
Intermittent CI Failure: Hyrax::XmlImportsController PATCH #update when files match metadata enqueues jobs for the matching file

### ISSUE THREAD

**Issue by mark-dce** [curationexperts/epigaea#752]
Commit: 8e07d6c8394b2368c0f041090c3c5e35a7ea1bc2
Seed: 61144

```

Failures:
  1) Hyrax::XmlImportsController PATCH #update when files match metadata enqueues jobs for the matching file
     
Failure/Error:
     
  expect { patch :update, params: params }
     
    .to enqueue_job(ImportJob)
     
    .exactly(:twice)
     
     
  expected to enqueue exactly 2 jobs, but enqueued 1
     
# ./spec/controllers/hyrax/xml_imports_controller_spec.rb:133:in `block (4 levels) in <top (required)>'
Finished in 8 minutes 55 seconds (files took 18.34 seconds to load)
1408 examples, 1 failure, 14 pending
Failed examples:
rspec ./spec/controllers/hyrax/xml_imports_controller_spec.rb:132
 
# Hyrax::XmlImportsController PATCH #update when files match metadata enqueues jobs for the matching file
Randomized with seed 61144
```

**Comment by no-reply** [comment]
See #768.

**Comment by val99erie** [comment]
I'm not sure if this is related or not, but...

There was a spec in `spec/presenters/xml_import_presenter_spec.rb` that was marked as intermittently failing, and it failed because a job wasn't getting queued when it was supposed to.  At one point I was able to get that spec to fail reliably in my dev env, so I turned on the debugger and found the problem.  It was because of the association between an `XmlImport` object and a `Batch` object.

At some point a list of IDs were being added to the batch object, but then when you called `batch.ids` from the XmlImport object, the result was an empty array.  The cause of the problem was that the in-memory batch object that the IDs were being added to had a different `object_id` than the batch object that the XmlImport object used to look up that same list of ids.  So there were 2 different in-memory objects that didn't have the same state.

I fixed that problem by adding the `inverse_of` property to the association between the 2 models. 
That code is currently in a PR:

https://github.com/curationexperts/epigaea/pull/806/commits/56380317abf53e07efb59305df7508104439a299

Since this is also an intermittent failure involving job queuing, I wonder if it's the same problem that I encountered, or if it's unrelated.
